app/models/stripe_record.rb

```diff
@@ -36,28 +48,33 @@ class StripeRecord < ApplicationRecord
   # Also saves us from some pains because JSON columns are highly inconsistent among MySQL and MariaDB.
   serialize :parameters, coder: JSON
 
+  def determine_wca_status
+    result = WCA_TO_STRIPE_STATUS_MAP.find { |key, values| values.include?(stripe_status) }
+    result&.first || raise("No associated wca_status for stripe_status: #{stripe_status} - our tests should prevent this from happening!")
+  end
+
   def find_account_id
     self.account_id || parent_transaction&.find_account_id
   end
 
   def update_status(api_transaction)
     stripe_error = nil
 
-    case self.api_type
+    case self.stripe_record_type
     when 'payment_intent'
       stripe_error = api_transaction.last_payment_error&.code
     when 'charge'
       stripe_error = api_transaction.failure_message
     end
 
-    self.update!(
-      status: api_transaction.status,
+    self.update(
```

Comment:
Subtle but important: You're changing from `update!` to `update` (without the exclamation sign)
Was this on purpose?